app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, :all
+    return unless user.present? # additional permissions for logged in users (they can read their own posts)
+
+    can :create, :all
+    can :manage, Post, author_id: user.id
+    can :manage, Comment, author_id: user.id
+    can :manage, Like, author_id: user.id
+    # can :read, Post, user: user
+    return unless user.is?(:admin) # additional permissions for administrators
+  end
+end
```

Comment:
- [ ] As an admin I should be able to delete a **post** or **comment**, whether it is mine or not, however, I am unable to do that. The user I assigned the admin role to cannot really delete other users' posts and comments. Kindly fix this.

  ## SUGGESTION
  You can consider adding the following code snippet to this method:
  ```
  ...

  if user.is?(:admin)
    can :manage, :all
  end

  ...
  ```